server/conn.go

```diff
@@ -133,20 +133,6 @@ var (
 		mysql.ComSetOption:        metrics.QueryTotalCounter.WithLabelValues("SetOption", "Error"),
 	}
 
-	queryDurationHistogramUse      = metrics.QueryDurationHistogram.WithLabelValues("Use")
```

Comment:
Is the removal of these lines intentional?

Reply:
@dveeden Yes, it is intentional. The reason to remove them is that they are fixed code with only 1 label of `sqlType` by default and it's okay because we only have a limited set of values for `sqlType` so hard code will work. But when we increase the labels with `db` and `table`, we have no way to hard coded these labels, it should be specified dynamically.

Reply:
@lysu Thanks for your comment so I understand why there so much hard code like this.

1. Is it a good idea that we define a config flag to enable/disable the metric? So user will have the right to choose between them like what we can do with `record-db-qps` config?
2. In case of we can not add the label `table` due to the high cardinalities problem, Is it okay that we still add label `db`?

Reply:
> @lysu Thanks for your comment so I understand why there so much hard code like this.
> 
> 1. Is it a good idea that we define a config flag to enable/disable the metric? So user will have the right to choose between them like what we can do with `record-db-qps` config?
> 2. In case of we can not add the label `table` due to the high cardinalities problem, Is it okay that we still add label `db`?

Adding a configuration for it like `record-db-qps` looks good to me.

Reply:
@lysu Is it okay that we define a new config flag and if it is turn off, the metrics of `handle_query_duration_seconds` does not appear, and when it is turn on the metrics register with 3 labels {sqlType, dbName, tableName} ?

Reply:
It is better to keep the original metrics as they used to be. There're many scripts of Grafana depending on it. It is better to add new metrics for the new feature. And the new config flag control the new metrics.